server/src/main/java/org/jboss/as/server/deployment/module/VFSResourceLoader.java

```diff
@@ -123,18 +139,41 @@ public Manifest run() throws IOException {
                 throw new UndeclaredThrowableException(e);
             }
         }
+        // "A multi-release jar file is a jar file that contains a manifest with a main attribute named "Multi-Release", [...]"
+        multiRelease = manifest != null && manifest.getMainAttributes().containsKey("Multi-Release");
```

Comment:
The Keys in the Attributes Map ar not Strings but Attirbutes#Names, so this should probably be containsKey(new Attributes.Name("Multi-Release"))